github-runner-manager/src/github_runner_manager/openstack_cloud/configuration.py

```diff
@@ -20,9 +23,21 @@ class OpenStackConfiguration:
     network: str
     credentials: "OpenStackCredentials"
 
+    @staticmethod
+    def from_yaml_file(file: TextIO) -> "OpenStackConfiguration":
+        """Initialize configuration from a YAML formatted file.
+
+        Args:
+            file: The file object to parse the configuration from.
+
+        Returns:
+            The configuration.
+        """
+        config = yaml.safe_load(file)
+        return OpenStackConfiguration.validate(config)
```

Comment:
I think this is no longer necessary.

```suggestion
    @staticmethod
    def from_yaml_file(file: TextIO) -> "OpenStackConfiguration":
        """Initialize configuration from a YAML formatted file.

        Args:
            file: The file object to parse the configuration from.

        Returns:
            The configuration.
        """
        config = yaml.safe_load(file)
        return OpenStackConfiguration.validate(config)
```